src/main/resources/assets/templates/product/productAdd.html

```diff
@@ -16,32 +16,32 @@ <h1 class="text-center">Create product</h1>
       <div class="col-lg-4 mt-4">
         <form class="row" th:object="${product}" th:action="@{/catalog/add}" method="post">
           <div class="mb-4">
-            <label class="form-label h5" th:for="username">Name</label>
+            <label class="form-label h5" th:for="name">Name</label>
             <input class="form-control" type="text" placeholder="Name" th:field="*{name}"
                    required>
           </div>
 
           <div class="mb-4">
-            <label class="form-label h5" th:for="firstName">Price</label>
-            <input class="form-control" type="text" placeholder="Price" th:field="*{price}"
+            <label class="form-label h5" th:for="price">Price</label>
+            <input class="form-control" type="number" step="0.01" placeholder="Price" th:field="*{price}"
```

Comment:
Does not work at front for float values

Reply:
Field error in object 'product' on field 'price': rejected value [0.05]; codes [typeMismatch.product.price,typeMismatch.price,typeMismatch.int,typeMismatch]; arguments [org.springframework.context.support.DefaultMessageSourceResolvable: codes [product.price,price]; arguments []; default message [price]]; default message [Failed to convert property value of type 'java.lang.String' to required type 'int' for property 'price'; nested exception is java.lang.NumberFormatException: For input string: "0.05"]]